### ISSUE TITLE
NullPointerException Thrown by Overriden Method

### ISSUE THREAD

**Issue by stain** [beanshell/beanshell#2]
Moved from https://code.google.com/a/apache-extras.org/p/beanshell/issues/detail?id=4

Reported by @pejobo, Apr 28, 2013
Originally from http://sourceforge.net/tracker/?func=detail&aid=2081602&group_id=4075&atid=104075

The following script throws `NullPointerException`.

``` java
import java.lang.Integer;
import java.sql.Timestamp;

public class A
{
public A (int id)
{
testMethod( "I" );
}

public void testMethod (String testString)
{
}

}

public class B extends A
{
public B ( int id )
{
super (id);
}

public void testMethod (String testString)
{
}
}

B mInOut = new B ( 0 );
```

By removing the testMethod() in constructor of A, it does not report anymore error. Why?

``` java
import java.lang.Integer;
import java.sql.Timestamp;

public class A
{
public A (int id)
{
// testMethod( "I" );
}

public void testMethod (String testString)
{
}

}

public class B extends A
{
public B ( int id )
{
super (id);
}

public void testMethod (String testString)
{
}
}

B mInOut = new B ( 0 );
```

**Comment by stain** [comment]
// Error: // Uncaught Exception: Typed variable declaration : Object constructor : at Line: 29 : in file: `<unknown file>` : `new B ( 0 ) `

Target exception: `bsh.InterpreterError`: Error in class initialization: `java.lang.NullPointerException`

```
bsh.InterpreterError: Error in class initialization: java.lang.NullPointerException
	at bsh.ClassGeneratorUtil.initInstance(ClassGeneratorUtil.java:1121)
	at A.<init>(BeanShell Generated via ASM (www.objectweb.org))
	at B.<init>(BeanShell Generated via ASM (www.objectweb.org))
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
	at bsh.Reflect.constructObject(Reflect.java:621)
	at bsh.BSHAllocationExpression.constructObject(BSHAllocationExpression.java:117)
	at bsh.BSHAllocationExpression.objectAllocation(BSHAllocationExpression.java:108)
	at bsh.BSHAllocationExpression.eval(BSHAllocationExpression.java:56)
	at bsh.BSHPrimaryExpression.eval(BSHPrimaryExpression.java:96)
	at bsh.BSHPrimaryExpression.eval(BSHPrimaryExpression.java:41)
	at bsh.BSHVariableDeclarator.eval(BSHVariableDeclarator.java:80)
	at bsh.BSHTypedVariableDeclaration.eval(BSHTypedVariableDeclaration.java:78)
	at bsh.Interpreter.run(Interpreter.java:471)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NullPointerException
	at B.testMethod(BeanShell Generated via ASM (www.objectweb.org))
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at bsh.Reflect.invokeMethod(Reflect.java:131)
	at bsh.BshMethod.invoke(BshMethod.java:216)
	at bsh.BshMethod.invoke(BshMethod.java:180)
	at bsh.Name.invokeLocalMethod(Name.java:911)
	at bsh.Name.invokeMethod(Name.java:798)
	at bsh.BSHMethodInvocation.eval(BSHMethodInvocation.java:69)
	at bsh.BSHPrimaryExpression.eval(BSHPrimaryExpression.java:96)
	at bsh.BSHPrimaryExpression.eval(bsh % BSHPrimaryExpression.java:41)
	at bsh.BSHBlock.evalBlock(BSHBlock.java:125)
	at bsh.BSHBlock.eval(BSHBlock.java:75)
	at bsh.BshMethod.invokeImpl(BshMethod.java:356)
	at bsh.BshMethod.invoke(BshMethod.java:252)
	at bsh.ClassGeneratorUtil.initInstance(ClassGeneratorUtil.java:1111)
	... 16 more
```

**Comment by pejobo** [comment]
See also https://github.com/pejobo/beanshell2/issues/45

**Comment by stain** [comment]
Thanks, @pejobo  - are you able to re-contribute your patch to this repository? I am unable to copy it myself as pejobo/beanshell is LGPL licensed, while contributiions to beanshell/beanshell will need to be covered by the Apache Software License. You are however the copyrightholder (presumably) and can do so. :)

**Comment by stain** [comment]
Hi, @pejobo - you originally had a patch for this bug in pejobo/beanshell2#45 - are you able to add this to the beanshell/beanshell master or as a pull request?

**Comment by nickl-** [comment]
beanshell2 fix merged at beanshell/beanshell@67e4aaa

added unit test [override2.bsh](https://github.com/beanshell/beanshell/blob/merge-fork-beanshell2/src/test/resources/test-scripts/override2.bsh)
